src/App.tsx

```diff
@@ -15,48 +17,129 @@ export const goodsFromServer = [
+          return good1[sortType].localeCompare(good2[sortType]);
+
+        case SortValue.Length:
+          return good1[sortType] - good2[sortType];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>({
+    sortType: null,
+    reverse: false,
+  });
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField);
+  const updateSortFieldKey = (key: keyof SortType, newValue?: SortValue) => {
+    setSortField((currentSortField) => {
+      const updatedSortField = { ...currentSortField };
+
+      if (key === 'reverse') {
+        updatedSortField[key] = !updatedSortField[key];
+      } else {
+        updatedSortField[key] = newValue as SortValue;
+      }
+
+      return updatedSortField;
+    });
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField.sortType !== SortValue.Name,
+          })}
+          onClick={() => updateSortFieldKey('sortType', SortValue.Name)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortField.sortType !== SortValue.Length,
+          })}
+          onClick={() => updateSortFieldKey('sortType', SortValue.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !sortField.reverse,
+          })}
+          onClick={() => updateSortFieldKey('reverse')}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField.sortType || sortField.reverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField({
+                sortType: null,
+                reverse: false,
+              });
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map((good) => (
+          <li data-cy="Good" key={good.id}>
+            {good.name}
+          </li>
+        ))}
```

Comment:
its better to use object destructuring

```suggestion
        {visibleGoods.map(({ id, name }) => (
          <li data-cy="Good" key={id}>
            {name}
          </li>
        ))}
```

Reply:
I don't think it's necessary